### ISSUE TITLE
Shouldn't `epoch_names_matching` return a set instead of an iterator?

### ISSUE THREAD

**Issue by bburan** [LBHB/NEMS0#88]
https://github.com/LBHB/NEMS/blob/3b6b62ded063d04e5684e819fd60c997e0959553/nems/epoch.py#L432

The current implementation returns an iterator. This is problematic for several reasons:

1. Anyone who attempts to iterate through the return value twice will find that on the second time nothing happens. That's because the iterator has been consumed. For example:
```
matches = epoch_names_matching(epoch, r'^TRIAL\d+')
for match in matches:
# do something
for match in matches:
# do another thing
```
The second for loop has nothing to iterate through since matches has been consumed.

2. The iterator does not return unique values. So, the same epoch can appear more than once in the result. 

It seems better to modify the function to return `set(filter)`. 

This is a simple fix but since this function is used throughout NEMS, @svdavid and @jacobpennington need to chime in on this first.

**Comment by svdavid** [comment]
I modified in in the normalization branch to return a list of unique matches. That was easier for me to work with than the filter. Shall I shove the normalization pull request through?

**Comment by bburan** [comment]
Sure. Yes, I saw a few minutes ago you had already added it. I don't have time to review the full pull request, but if you think it's good I see no reason to pull the whole thing into master.

Brad

________________________________

**Comment by svdavid** [comment]
I think this is now fixed with #75
